src/aws-cpp-sdk-core/source/client/AWSClient.cpp

```diff
@@ -289,7 +289,7 @@ HttpResponseOutcome AWSClient::AttemptExhaustively(const Aws::Http::URI& uri,
                                                                false/*retryable*/));
 
         };
-        httpRequest->SetEventStreamRequest(request.IsEventStreamRequest());
+        httpRequest->SetEventStreamRequest(request.IsEventStreamRequest() || request.HasEventStreamResponse());
```

Comment:
Please don't overload `IsEventStreamRequest` boolean in the HttpRequest object, it may create unwanted side-effects of how the request is sent.
Instead, create a new boolean flag at set it accordingly.